mesa/agent.py

```diff
@@ -57,3 +60,41 @@ def advance(self) -> None:
     @property
     def random(self) -> Random:
         return self.model.random
+
+
+class AgentSet:
+    def __init__(self, agents: set[Agent], model: Model):
+        self._agents = agents
+        self.model = model
```

Comment:
why do you need a model attribute? I guess for model.random, but then can't you just pass the reference to the seeded random number generator itself?

Reply:
Good question. Let me think about it and possible cases.

Reply:
This way it’s explicit we use the same `model.random` thing as elsewhere. Are there disadvantages about including it / advantages of removing?

Reply:
Ok, let's mimic how Agent does it.